packages/core/content-releases/server/src/controllers/release-action.ts

```diff
@@ -17,6 +20,62 @@ const releaseActionController = {
       data: releaseAction,
     };
   },
+  async findMany(ctx: Koa.Context) {
+    const releaseId: GetReleaseActions.Request['params']['releaseId'] = ctx.params.releaseId;
+
+    const allowedContentTypes = getAllowedContentTypes({ strapi, userAbility: ctx.state.userAbility });
+    
+    const morphSanitizedPopulate = await allowedContentTypes.reduce(async (promiseAccumulator, contentTypeUid) => {
+      const accumulator = await promiseAccumulator;
+
+      const permissionsManager = await strapi.admin.services.permission.createPermissionsManager({
+        ability: ctx.state.userAbility,
+        model: contentTypeUid
+      });
+
+      // We create this to filter populated entries based on custom conditions on permissions
+      accumulator[contentTypeUid] = {
+        filters: permissionsManager.getQuery(CONTENT_MANAGER_READ_ACTION)
+      };
+
+      return accumulator;
+    }, Promise.resolve({} as Record<UID.ContentType, { filters: object }>));
+
+
+    const releaseService = getService('release', { strapi });
+    const { results, pagination } = await releaseService.findActions(releaseId, allowedContentTypes, { 
+      populate: {
+        entry: {
+          on: morphSanitizedPopulate
+        }
+      }
+    });
+
+    // Because this is a morphTo relation, we need to sanitize each entry separately based on its contentType
+    const sanitizedResult = await Promise.all(results.map(async (action: ReleaseAction) => {
+      const permissionsManager = strapi.admin.services.permission.createPermissionsManager({
+        ability: ctx.state.userAbility,
+        model: action.contentType
+      });
+
+      return {
+        ...action,
+        entry: action.entry && await permissionsManager.sanitizeOutput(
+          action.entry, 
+          { 
+            action: CONTENT_MANAGER_READ_ACTION, 
+            subject: permissionsManager.toSubject(action.entry, action.contentType) 
+          })
+      };
+    }));
```

Comment:
I feel like there should be a better way to handle this, because right now I'm creating a permissions manager for each action.

Even if this is the right way to go, I would like to change this to try to memorize the permissionsManager, so I don't create a permissionsManager already used

Reply:
I believe the permission manager should work with polymorphic relations.

Would it be possible to work with the release instead of actions?
```
      const permissionsManager = strapi.admin.services.permission.createPermissionsManager({
        ability: ctx.state.userAbility,
        model: ReleaseUID
      });
      
	permissionsManager.sanitizeOutput(
		release.entry,
		{ ... }
	)
 ``` 
 
 That way the sanitizer would automatically filter the actions the user doesnt have permissions to read.